src/templates/areaDeTrabalho/components/AvaliarCadernos.tsx

```diff
@@ -44,19 +76,51 @@ const AvaliarCadernos = ({ idvol }: AvaliarCadernosProps) => {
       }
       return notebook;
     });
-
-    setNotebooksIn(updatedNotebooks);
+    const filterReserved = sortedNotebooksReserved(
+      updatedNotebooks.filter((notebook) => notebook.reserved)
+    );
+    const filterNotReserved = sortedNotebooksNotReserved(
+      updatedNotebooks.filter((notebook) => !notebook.reserved)
+    );
+    setNotebooksIn([filterReserved, filterNotReserved].flat());
 
     await putReservationData(notebookId);
   };
 
+  const handleRevertReservation = async (notebookId: number) => {
+    const updatedNotebooks = notebooksIn.map((notebook) => {
```

Comment:
Esse trecho também pode ser modularizado para:
```
const updateNotebooksReservation = (notebookId: number, reserveFlag: boolean) => {
    return notebooksIn.map((notebook) => {
      if (notebook.notebookId === notebookId) {
        return {
          ...notebook,
          reserved: reserveFlag,
        };
      }
      return notebook;
    });
  }
```

e ser chamado com `const updatedNotebooks = updateNotebooksReservation(notebookId, true);` ou `const updatedNotebooks = updateNotebooksReservation(notebookId, false);` dependendo de onde for chamado